pkg/ee/resource-quota/default-quota-controller/controller.go

```diff
@@ -111,83 +114,71 @@ func (r *reconciler) reconcile(ctx context.Context, setting *kubermaticv1.Kuberm
 		return fmt.Errorf("failed to list resource quotas: %w", err)
 	}
 
-	projectQuotas := pairProjectQuotas(projects, resourceQuotas)
-	// if the setting is nil, remove all default resource quotas
+	// Delete all default project quotas if user didn't specify any default project quota.
 	if setting.Spec.DefaultProjectResourceQuota == nil {
-		if err := r.handleDeletion(ctx, projectQuotas); err != nil {
-			return fmt.Errorf("error deleting default project quotas %w", err)
-		}
-		return nil
+		return r.handleDeletion(ctx)
 	}
 
-	for _, pQuota := range projectQuotas {
-		if err := r.ensureDefaultProjectQuota(ctx, setting, &pQuota.project, pQuota.quota); err != nil {
-			return fmt.Errorf("error ensuring default project quotas: %w", err)
+	return r.synchronizeResourceQuotas(ctx, setting.Spec.DefaultProjectResourceQuota, projects, resourceQuotas)
+}
+
+func (r *reconciler) synchronizeResourceQuotas(ctx context.Context, defaultResourceQuota *kubermaticv1.DefaultProjectResourceQuota, projects *kubermaticv1.ProjectList, quotas *kubermaticv1.ResourceQuotaList) error {
+	// We need to synchronize default resource quotas.
+	var defaultResourceQuotas []reconciling.NamedResourceQuotaReconcilerFactory
+
+	// Create a lookup for projects with resource quotas.
+	resourceQuotaLookup := map[string]kubermaticv1.ResourceQuota{}
+	for _, quota := range quotas.Items {
+		if quota.Spec.Subject.Kind == kubermaticv1.ProjectSubjectKind {
+			resourceQuotaLookup[quota.Spec.Subject.Name] = quota
 		}
 	}
 
-	return nil
-}
+	// Iterate over all the projects and synchronize projects with their default quotas.
+	for _, project := range projects.Items {
+		// Ignore projects that are queued for deletion.
+		if project.DeletionTimestamp == nil {
+			continue
+		}
```

Comment:
This should be `!=`

Reply:
You see, that's exactly why I asked for _**your**_ review.

Reply:
If you hadn't told me to take a deeper look, I would have not seen this. Don't give me too much credit.